container/build_tar.py

```diff
@@ -178,6 +192,32 @@ def add_link(self, symlink, destination):
     symlink = os.path.normpath(symlink)
     self.tarfile.add_file(symlink, tarfile.SYMTYPE, link=destination)
 
+  @contextmanager
+  def write_temp_file(self, data, suffix='tar', mode='wb'):
```

Comment:
Can you use tempfile.TemporaryFile here instead of making your own context manager?
https://docs.python.org/2/library/tempfile.html#tempfile.TemporaryFile

Reply:
I tried using Temporary file, but unfortunately i cant. 
This temporary file is then opened using tarfile.open L:207. However, tar_open throws an exception, if we try to open a file already opened for reading and writing. 
I cannot avoid using tarfile.open since, i want to extract a file from the binary blob.